src/main/scala/scorex/transaction/SimpleTransactionModule.scala

```diff
@@ -98,7 +98,8 @@ class SimpleTransactionModule(hardForkParams: ChainParameters)(implicit val sett
     val txs = utxStorage.all()
     val notExpired = txs.filter { tx => (currentTime - tx.timestamp).millis <= MaxTimeForUnconfirmed }
     val notFromFuture = notExpired.filter { tx => (tx.timestamp - currentTime).millis <= MaxTimeDrift }
-    val valid = blockStorage.state.validate(notFromFuture, blockTime = currentTime)
```

Comment:
Have you considered `UnconfirmedTransactionsDatabaseImpl.transactions` to be a `Sorted` structure so adding/removing wouldn't require explicit resorting?